inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/config/model/instrumentation/dataproviders/GenericDataProviderSettings.java

```diff
@@ -21,19 +21,30 @@
 public class GenericDataProviderSettings {
 
     public static final String PACKAGE_REGEX = "[a-zA-Z]\\w*(\\.[a-zA-Z]\\w*)*";
+
     public static final String THIZ_VARIABLE = "thiz";
     public static final String ARGS_VARIABLE = "args";
     public static final String THROWN_VARIABLE = "thrown";
     public static final String RETURN_VALUE_VARIABLE = "returnValue";
     public static final String ARG_VARIABLE_PREFIX = "arg";
     public static final Pattern ARG_VARIABLE_PATTERN = Pattern.compile(ARG_VARIABLE_PREFIX + "(\\d+)");
```

Comment:
This is stored as compiled pattern, because it is used for regex-matching in the DataProviderGenerator. I think storing a regex as public-static-final makes little sense as you will only use it in compiled form anyway.